File: ls_shop/lifestyle_shop_ecommerce/dashboard_chart_source/facebook_pageview_trend/facebook_pageview_trend.py
import frappe

from ls_shop.analytics import facebook


@frappe.whitelist()
def get(
	chart_name: str | None = None,
	chart: dict | str | None = None,
	no_cache: int | None = None,
	filters: dict | list | str | None = None,
	from_date: str | None = None,
	to_date: str | None = None,
	timespan: str | None = None,
	time_interval: str | None = None,
	heatmap_year: str | int | None = None,
	refresh: int | None = None,
):
	frappe.only_for("System Manager")
	# broken credentials must render an empty chart, not error the Desk dashboard
	try:
		daily = facebook.get_summary()["daily_pageviews"]
	except Exception:
		daily = {}
	return {
		"labels": list(daily),
		"datasets": [{"name": "PageView", "values": list(daily.values())}],
	}

File: ls_shop/lifestyle_shop_ecommerce/dashboard_chart_source/facebook_pageview_trend/facebook_pageview_trend.js
frappe.provide('frappe.dashboards.chart_sources');

frappe.dashboards.chart_sources['Facebook PageView Trend'] = {
	method:
		'ls_shop.lifestyle_shop_ecommerce.dashboard_chart_source.facebook_pageview_trend.facebook_pageview_trend.get',
	filters: [],
};
